networkx/algorithms/perfect_graph.py

```diff
@@ -0,0 +1,80 @@
+import networkx as nx
+from networkx.exception import NetworkXError
+
+__all__ = ["is_perfect_graph"]
+
+
+def is_perfect_graph(G, *, max_cycle_length=None):
```

Comment:
I'm inclined to remove the `max_cycle_length`. I understand the desire to add it for performance reasons, but technically `is_perfect_graph` is a misnomer when `max_cycle_length` is anything other than `None`.

I also generally am -1 on the pattern of warning when checking for many cycles, which could naturally go away if we got rid of this kwarg.